test/extended/storage/driver_configuration.go

```diff
@@ -0,0 +1,326 @@
+		snapshotOptions          = map[string]map[string]string{
+			"global": {
+				"clusterCSIDriver": "globalMaxSnapshotsPerBlockVolume",
+				"cloudConf":        "global-max-snapshots-per-block-volume",
+			},
+			"vsan": {
+				"clusterCSIDriver": "granularMaxSnapshotsPerBlockVolumeInVSAN",
+				"cloudConf":        "granular-max-snapshots-per-block-volume-vsan",
+			},
+			"vvol": {
+				"clusterCSIDriver": "granularMaxSnapshotsPerBlockVolumeInVVOL",
+				"cloudConf":        "granular-max-snapshots-per-block-volume-vvol",
+			},
+		}
+	)
+
+	g.BeforeEach(func() {
+		//TODO: remove when GA
+		if !exutil.IsTechPreviewNoUpgrade(oc) {
+			g.Skip("this test is only expected to work with TechPreviewNoUpgrade clusters")
+		}
+
+		if !framework.ProviderIs("vsphere") {
+			g.Skip("this test is only expected to work with vSphere clusters")
+		}
+
+		originalClusterCSIDriver, err := oc.AdminOperatorClient().OperatorV1().ClusterCSIDrivers().Get(context.Background(), providerName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		originalDriverConfigSpec = originalClusterCSIDriver.Spec.DriverConfig.DeepCopy()
+		e2e.Logf("storing original driver config: %+v", originalDriverConfigSpec)
+	})
+
+	g.AfterEach(func() {
+		if originalDriverConfigSpec == nil {
+			return
+		}
+
+		clusterCSIDriver, err := oc.AdminOperatorClient().OperatorV1().ClusterCSIDrivers().Get(context.Background(), providerName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("restoring original driver config: %+v", originalDriverConfigSpec)
+		clusterCSIDriver.Spec.DriverConfig = *originalDriverConfigSpec
+		_, err = oc.AdminOperatorClient().OperatorV1().ClusterCSIDrivers().Update(context.Background(), clusterCSIDriver, metav1.UpdateOptions{})
+		if err != nil {
+			e2e.Failf("failed to update ClusterCSIDriver: %v", err)
+		}
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Context("snapshot options in clusterCSIDriver", func() {
+		var tests = []struct {
+			name                       string
+			clusterCSIDriverOptions    map[string]int
+			cloudConfigOptions         map[string]int
+			successfulSnapshotsCreated int // Number of snapshots that should be created successfully, 0 to skip.
+		}{
+			{
+				name: "global snapshot option",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["global"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["global"]["cloudConf"]: 4,
+				},
+				successfulSnapshotsCreated: 4,
+			},
+			{
+				name:                       "default snapshot limit",
+				clusterCSIDriverOptions:    map[string]int{},
+				cloudConfigOptions:         map[string]int{},
+				successfulSnapshotsCreated: 3,
+			},
+			{
+				name: "vsan snapshot option",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["vsan"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["vsan"]["cloudConf"]: 4,
+				},
```

Comment:
Why so complicated?

```
clusterCSIDriverOptions: opv1.VSphereCSIDriverConfigSpec {
    granularMaxSnapshotsPerBlockVolumeInVSAN: ptr.To(uint32(4)),
},

cloudConfigOptions: map[string]string{
    "granular-max-snapshots-per-block-volume-vsan": "4".
}
```

(note change of the field types)

It will be slightly harder to compute the ClusterCSIDriver patch, but you can use `Update()` instead or encode VSphereCSIDriverConfigSpec to json.
On the positive side, it's crystal clear what field is set and what ini file is expected without reading a separate map.

Reply:
Changed and bumping openshift/api to pull in the new clusterCSIDriver fields.